polaris/experimental/_dataset_v2.py

```diff
@@ -0,0 +1,267 @@
+        """
+        self.client.upload_dataset(self, access=access, owner=owner)
+
+    def to_json(
+        self,
+        destination: str,
+        if_exists: ZarrConflictResolution = "replace",
+        load_zarr_from_new_location: bool = False,
+    ) -> str:
+        """
+        Save the dataset to a destination directory as a JSON file.
+
+        Args:
+            destination: The _directory_ to save the associated data to.
+            if_exists: Action for handling existing files in the Zarr archive. Options are 'raise' to throw
+                an error, 'replace' to overwrite, or 'skip' to proceed without altering the existing files.
+            load_zarr_from_new_location: Whether to update the current instance to load data from the location
+                the data is saved to. Only relevant for Zarr-datasets.
+
+        Returns:
+            The path to the JSON file.
+        """
+        destination = Path(destination)
+        destination.mkdir(exist_ok=True, parents=True)
+
+        dataset_path = str(destination / "dataset.json")
+        new_zarr_root_path = str(destination / "data.zarr")
+
+        # Lu: Avoid serilizing and sending None to hub app.
+        serialized = self.model_dump(exclude={"cache_dir"}, exclude_none=True)
+        serialized["zarrRootPath"] = new_zarr_root_path
+
+        # Copy over Zarr data to the destination
+        self._warn_about_remote_zarr = False
+
+        logger.info(f"Copying Zarr archive to {new_zarr_root_path}. This may take a while.")
+        dest = zarr.open(new_zarr_root_path, "w")
+
+        zarr.copy_store(
+            source=self.zarr_root.store.store,
+            dest=dest.store,
+            log=logger.debug,
+            if_exists=if_exists,
+        )
+
+        if load_zarr_from_new_location:
+            self.zarr_root_path = new_zarr_root_path
+            self._zarr_root = None
+            self._zarr_data = None
+
+        with fsspec.open(dataset_path, "w") as f:
+            json.dump(serialized, f)
+        return dataset_path
+
+    def cache(self) -> str:
+        """Caches the dataset by downloading all additional data for pointer columns to a local directory.
+
+        Args:
+            verify_checksum: Whether to verify the checksum of the dataset after caching.
+
+        Returns:
+            The path to the cache directory.
+        """
+        # NOTE (cwognum): We don't support a deterministic checksum for the Dataset V2 yet,
+        #  so verification doesn't make sense. See also:
+        #  https://github.com/polaris-hub/polaris/issues/188
+        super().cache(verify_checksum=False)
+
+    def __getitem__(self, item):
+        """Allows for indexing the dataset directly"""
+        raise NotImplementedError
+
+    def _repr_dict_(self) -> dict:
+        """Utility function for pretty-printing to the command line and jupyter notebooks"""
+        repr_dict = self.model_dump(exclude={"zarr_md5sum_manifest"})
```

Comment:
**Question**
Since `zarr_md5sum_manifest`is a `PrivateAttr`, do we need to exclude it explicitly here?